.github/workflows/handle-closed-prs.yml

```diff
@@ -0,0 +1,67 @@
+name: Remove PR Previews
+
+on:
+  pull_request:
+    types: [closed]
+
+concurrency: preview-teardown-${{ github.ref }}
+
+jobs:
+  list-environments:
+    runs-on: ubuntu-latest
+    outputs:
+      hasNextJsEnv: ${{ steps.list-environments.outputs.hasNextJsEnv }}
+      hasVanillaJsEnv: ${{ steps.list-environments.outputs.hasVanillaJsEnv }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: List repo environments
+        id: list-environments
+        uses: actions/github-script@v6
+        env:
+          GITHUB_TOKEN: ${{ secrets.PIE_BOT_TOKEN }}
+        with:
+          script: |
+            const pullRequestNumber = context.issue.number;
+            const owner = context.repo.owner;
+            const repo = context.repo.repo;
+
+            // List environments
+            const environments = await github.rest.repos.getAllEnvironments({
+              owner: owner,
+              repo: repo,
+            });
+
+            const envList = environments.data.environments.map(env => env.name);
+
+            // Check for existence of specific environments
+            const hasNextJsEnv = envList.includes(`nextjs-app-pr-${pullRequestNumber}`);
+            const hasVanillaJsEnv = envList.includes(`vanilla-app-pr-${pullRequestNumber}`);
+
+            core.setOutput('hasNextJsEnv', hasNextJsEnv);
+            core.setOutput('hasVanillaJsEnv', hasVanillaJsEnv);
```

Comment:
Another thing to note for the infra-as-code epic is pulling out these large blocks into their own JS modules and writing tests for them